addons/website_sale/models/website.py

```diff
@@ -43,14 +43,13 @@ def _default_recovery_mail_template(self):
     shop_ppg = fields.Integer(default=20, string="Number of products in the grid on the shop")
     shop_ppr = fields.Integer(default=4, string="Number of grid columns on the shop")
 
-    @api.one
     def _compute_pricelist_ids(self):
-        """ Return the pricelists that can be used directly or indirectly on
-        the website.
-        """
-        Pricelist = self.env["product.pricelist"]
-        domain = Pricelist._get_website_pricelists_domain(self.id)
-        self.pricelist_ids = Pricelist.search(domain)
+        Pricelist = self.env['product.pricelist']
+        for website in self:
+            # TODO: potential performance bottleneck, use read_group ?
```

Comment:
As far as I know, you don't often do stuff on *batches* of websites. Having 10 websites in the table is already pretty rare...